src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1116,6 +1181,106 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
+								IntrinsicId.Type_GetFields => DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.NonPublicFields,
+								IntrinsicId.Type_GetProperties => DynamicallyAccessedMemberTypes.PublicProperties | DynamicallyAccessedMemberTypes.NonPublicProperties,
+								IntrinsicId.Type_GetNestedTypes => DynamicallyAccessedMemberTypes.PublicNestedTypes | DynamicallyAccessedMemberTypes.NonPublicNestedTypes,
+								IntrinsicId.Type_GetMembers => DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors |
+									DynamicallyAccessedMemberTypes.PublicEvents | DynamicallyAccessedMemberTypes.NonPublicEvents |
+									DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.NonPublicFields |
+									DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods |
+									DynamicallyAccessedMemberTypes.PublicProperties | DynamicallyAccessedMemberTypes.NonPublicProperties |
+									DynamicallyAccessedMemberTypes.PublicNestedTypes | DynamicallyAccessedMemberTypes.NonPublicNestedTypes,
+								_ => throw new ArgumentException ($"Reflection call '{calledMethod.GetDisplayName ()}' inside '{callingMethodDefinition.GetDisplayName ()}' is of unexpected member type."),
+							};
+						} else {
+							memberTypes = callType switch
+							{
+								IntrinsicId.Type_GetConstructors => GetDynamicallyAccessedMemberTypesFromBindingFlagsForConstructors (bindingFlags),
+								IntrinsicId.Type_GetMethods => GetDynamicallyAccessedMemberTypesFromBindingFlagsForMethods (bindingFlags),
+								IntrinsicId.Type_GetEvents => GetDynamicallyAccessedMemberTypesFromBindingFlagsForEvents (bindingFlags),
+								IntrinsicId.Type_GetFields => GetDynamicallyAccessedMemberTypesFromBindingFlagsForFields (bindingFlags),
+								IntrinsicId.Type_GetProperties => GetDynamicallyAccessedMemberTypesFromBindingFlagsForProperties (bindingFlags),
+								IntrinsicId.Type_GetNestedTypes => GetDynamicallyAccessedMemberTypesFromBindingFlagsForNestedTypes (bindingFlags),
+								IntrinsicId.Type_GetMembers => GetDynamicallyAccessedMemberTypesFromBindingFlagsForMembers (bindingFlags),
+								_ => throw new ArgumentException ($"Reflection call '{calledMethod.GetDisplayName ()}' inside '{callingMethodDefinition.GetDisplayName ()}' is of unexpected member type."),
+							};
+						}
+
+						foreach (var value in methodParams[0].UniqueValues ()) {
+							RequireDynamicallyAccessedMembers (ref reflectionContext, memberTypes, value, calledMethodDefinition);
+						}
+					}
+					break;
+
+
+				//
+				// GetMember (String)
+				// GetMember (String, BindingFlags)
+				// GetMember (String, MemberTypes, BindingFlags)
+				//
+				case IntrinsicId.Type_GetMember when calledMethod.DeclaringType.Namespace == "System"
+					&& calledMethod.DeclaringType.Name == "Type"
+					&& calledMethod.HasParameterOfType (0, "System", "String"): {
```

Comment:
This doesn't make sense to me - we check exactly this in the first switch - so no reason to check the namespace, typename and first parameter here again. My comment was about the fact that the code below checks that BindingFlags parameter is in a certain position based on the number of parameters to the method. And if those checks don't succeed we fail. But the switch above which detects the `IntrinsicId` does not perform those checks.

So for example, if there was a method Type.GetMember(String, MemberTypes)  (no binding flags), the first switch would accept it as recognized and detected it as `IntrinsicId.Type_GetMember`, it would get here, and the code below would fail because the binding flags checks would fail - it would end up effectively crashing the linker. That's not good. The check which detects the `IntrinsicId` needs to be very precise, so that the code in here (HandleCall) can't ever fail (well at least for reasons of not recognizing the method).